Winleafs.Wpf/Views/Layout/CreateEffectWindow.xaml.cs

```diff
@@ -0,0 +1,262 @@
+				{
+					color = prevColor;
+				}
+
+				newFrame.PanelColors.Add(panelId, color);
+            }                
+
+			// Add the frame to the effect and the displayed list and select it
+            _customEffect.Frames.Add(newFrame);
+			var newFrameListItem = new FrameListItem(newFrame, string.Format(_frameNameFormat, _frameListItems.Count + 1));
+			_frameListItems.Add(newFrameListItem);
+			FrameListBox.SelectedItem = newFrameListItem;
+        }
+
+		private void BuildFrameList()
+		{
+			_frameListItems = new ObservableCollection<FrameListItem>();
+
+			for (var i = 0; i < _customEffect.Frames.Count; i++)
+			{
+				_frameListItems.Add(new FrameListItem(_customEffect.Frames[i], string.Format(_frameNameFormat, i + 1)));
+			}
+
+			FrameListBox.ItemsSource = _frameListItems;
+
+			//Always show at least 1 frame in the list
+			if (_customEffect.Frames.Count == 0)
+			{
+				AddNewFrame();
+			}
+
+			FrameListBox.SelectedItem = _frameListItems.First();
+		}
+
+		private void BuildRgbToBrushMap()
+		{
+			//Get the colours used across all teh frames, and make sure black (not lit) is included
+			var colorsUsed = _customEffect.Frames.SelectMany(f => f.PanelColors.Values).Append(0U).Distinct();
+
+			//Add the colors used to the color picker control
+			ColorPicker.StandardColors.Clear();
+			_rgbToBrushMap = new Dictionary<uint, SolidColorBrush>();
+			
+			//Create a map of color to brush so that brushes can be reused
+			foreach (var rgb in colorsUsed)
+			{
+				AddToRgbToBrushMap(ColorFormatConverter.ToMediaColor(rgb));
+			}
+		}
+
+		private void AddToRgbToBrushMap(Color color)
+		{
+			ColorPicker.StandardColors.Add(new Xceed.Wpf.Toolkit.ColorItem(color, string.Empty));
+			_rgbToBrushMap.Add(ColorFormatConverter.ToRgb(color), color == Colors.Black ? Brushes.LightSlateGray : new SolidColorBrush(color));
+		}
+
+        private void Cancel_Click(object sender, RoutedEventArgs e)
+        {
+			//TODO Check for losing changes
+            Close();
+        }
+
+        private void SaveProgress_Click(object sender, RoutedEventArgs e)
+        {
+            UserSettings.Settings.ActiveDevice.CustomEffect = _customEffect;
+            UserSettings.Settings.SaveSettings();
+        }
+
+		private async void PlayOnDevice_Click(object sender, RoutedEventArgs e)
+		{
+			if (!float.TryParse(TransitionTextBox.Text, out var transitionSecs))
+			{
+				PopupCreator.Error(Layout.Resources.ValidTransistionTime);
+				return;
+			}
+
+			_customEffect.IsLoop = LoopCheckBox.IsChecked ?? false;
+			var customEffectCommandBuilder = new CustomEffectCommandBuilder(_customEffect);
+			var customEffectCommand = customEffectCommandBuilder.BuildDisplayCommand(transitionSecs);
+
+			var orchestrator = OrchestratorCollection.GetOrchestratorForDevice(UserSettings.Settings.ActiveDevice);
+
+			//TODO async so as not hang UI thread, but consider ContinueWith to handle errors
```

Comment:
TODO

Reply:
Also to be honest, this is fine for now. Async void is not great and we do know as much but the workarounds aren't pretty either. If you know something that would be nice, we would for sure be down to try it. Otherwise, it doesn't matter as much.

Reply:
I had a look at the code for calling the API and it looked like it was logging and then swallowing the exception rather than raising it , so I would say there isn't much to worry about with the exception handling here, so I may as well just remove the comment.